components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1083,6 +1083,11 @@ void RewardsServiceImpl::SetRewardsMainEnabled(bool enabled) const {
   return ledger_->SetRewardsMainEnabled(enabled);
 }
 
+void RewardsServiceImpl::GetRewardsMainEnabled() {
+  bool rewards_main_enabled = ledger_->GetRewardsMainEnabled();
+  TriggerOnRewardsMainEnabled(rewards_main_enabled);
```

Comment:
observer methods should never be triggered in response to a query on the rewards service. If we need an event for this it should go on SetRewardsMainEnabled